documentation/src/docs/metrics.md

```diff
@@ -7,553 +7,448 @@ You may obtain a copy of the License at
+     ...
+     dl.incCounterValue(SEGMENT_WRITE_BYTES, dataLength, segmentTags(streamSegmentName));
+    ```
+Here `SEGMENT_WRITE_BYTES` is the base name of the metric. Below are the two metrics associated with it:
+
+- The global Counter which has no tags associated.
+- A Segment specific Counter which has a list of Segment tags associated.
+
+Note that, the `segmentTags` is a method to generate tags based on fully qualified Segment name.
+
+The following are the required steps to register and use `OpStatsLogger(Timer)`:
+
+
+1. Get a `StatsLogger` from `MetricsProvider`.
+
+   ```
+     StatsLogger STATS_LOGGER = MetricsProvider.getStatsLogger("segmentstore");
+   ```
+2. Register all the desired metrics through `StatsLogger`.
+   ```
+     @Getter(AccessLevel.PROTECTED)
+     final OpStatsLogger createStreamSegment = STATS_LOGGER.createStats(SEGMENT_CREATE_LATENCY);
+   ```
+3. Use these metrics within code at the appropriate places where the values should be collected and recorded.
+
+   ```
+     getCreateStreamSegment().reportSuccessEvent(elapsed);
+   ```
+ Here `SEGMENT_CREATE_LATENCY` is the name of this metric, and `createStreamSegment` is the metric object, which tracks operations of `createSegment` and we will get the time (i.e. time taken by each operation and other numbers computed based on them) for each `createSegment` operation happened.
+
+## Example for Dynamic Gauge
+
+This is an example from `io.pravega.controller.metrics.StreamMetrics`. In this class, we report
+a Dynamic Gauge which represents the open Transactions of a Stream. The code for this example can be found [here](https://github.com/pravega/pravega/blob/master/controller/src/main/java/io/pravega/controller/metrics/StreamMetrics.java).
+
+## Example for Dynamic Meter
+
+This is an example from `io.pravega.segmentstore.server.SegmentStoreMetrics`. The code for this example can be found [here](https://github.com/pravega/pravega/blob/master/segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java). In the class `SegmentStoreMetrics`, we report a Dynamic Meter which represents the Segments created with a particular container.
+
+
+# Metric Registries and Configurations
+
+With Micrometer, each meter registry is responsible for both storage and exporting of metrics objects.
+In order to have a unified interface, Micrometer provides the `CompositeMeterRegistry` for the application to interact with, `CompositeMeterRegistry` will forward metric operations to all the concrete registries bounded to it.
+
+Note that `CompositeMeterRegistry` has no storage associated, hence in case no registry is bound to it, it will become an `NO-OP` interface only and Pravega will throw errors in such a case.
+
+For performing unit testing in an easier way, Micrometer also provides `SimpleMeterRegistry`, which has memory only storage but no exporting is allowed; call `startWithoutExporting()` of `StatsProvider` to use this feature in test codes.
+
```

Comment:
@kevinhan88 can u rephrase the above two paras...meaning is not clear.
`Note that `CompositeMeterRegistry` has no storage associated, hence in case no registry is bound to it, it will become an `NO-OP` interface only and Pravega will throw errors in such a case.` and `For performing unit testing in an easier way, Micrometer also provides `SimpleMeterRegistry`, which has memory only storage but no exporting is allowed; call `startWithoutExporting()` of `StatsProvider` to use this feature in test codes.`

Reply:
Note that when metrics service starts, only a global registry (of type CompositeMeterRegistry) is provided, which will bind concrete registries depends on the config. If no registry is switched on, metrics service throws error to prevent the global registry runs into no-op mode.

Mainly for testing purpose, metrics service can also startWithoutExporting(), where a SimpleMeterRegistry is bound. SimpleMeterRegistry holds memory only storage and does not export metrics, which is ideal for unit test to verify metrics objects.

Reply:
Accepted suggestion